events/start_handler.go

```diff
@@ -64,21 +69,25 @@ func (h *StartHandler) Handle(event *docker.APIEvents) error {
 	return nil
 }
 
-func (h *StartHandler) getRancherIP(c *docker.Container) string {
+func (h *StartHandler) getRancherIP(c *docker.Container) (string, error) {
 	for _, env := range c.Config.Env {
 		if strings.HasPrefix(env, RancherIPKey) {
-			return strings.TrimPrefix(env, RancherIPKey)
+			return strings.TrimPrefix(env, RancherIPKey), nil
 		}
 	}
 
 	filePath := path.Join(getContainerStateDir(), c.ID)
 	if _, err := os.Stat(filePath); err == nil {
+		time.Sleep(time.Millisecond * 200)
 		file, e := ioutil.ReadFile(filePath)
 		if e != nil {
-			log.Errorf("Error reading file for container %s: %v", c.ID, e)
+			return "", fmt.Errorf("Error reading file for container %s: %v", c.ID, e)
 		}
 		var instanceData instance
-		json.Unmarshal(file, &instanceData)
```

Comment:
I was accidentally eating the json unmarshall error. So, no longer eating it and bubbling up any error encountered while trying to read the file or unmarshall the json.